public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,171 @@
+# dvc.api.open()
+
+[Context manager](https://www.python.org/dev/peps/pep-0343/#context-managers-in-the-standard-library)
+to open a tracked file.
+
+## Definition
+
+### Signature
+
+```py
+open(path, repo=None, rev=None, remote=None, mode="r", encoding=None)
+```
+
+### Types
+
+All **parameter** types are
+[string](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str).
+
+The **return** type is a
+[file object](https://docs.python.org/3/glossary.html#term-file-object).
+
+No exceptions are raised by this function directly.
+
+## Description
+
+This function is analogous to the
+[`open()`](https://docs.python.org/3/library/functions.html#open) Python
+builtin, but for files tracked in <abbr>DVC projects</abbr> (by DVC or Git).
+However, it may only be used as a _context manager_ (using the `with` keyword as
+shown in the **Examples** below). There is no `close()`.
```

Comment:
not sure I understand this ... no `close()` where ... what does it call on the context exit?

Reply:
The paragraph explains an analogy between this fn and the builtin `open()`, however the builtin is not a context manager, instead it expects you will later `close()` the file. That's what this last sentence refers to.

> what does it call on the context exit?

Internally it probably calls some sort of close but this is not something we need to document in the api ref, I think.

Changed to ``There is no `dvc.api.close()`.`` to make it more obvious for now.

Reply:
so, regular open() is context manager (or something that behaves exactly like it, since it's a built-in function I'm not sure about exact implementation and it's not that important). So, there is a direct analogy. I would even first go first with a simple paragraph that we can take from a regular open. After that specify that it can be used with `with`.

Reply:
OK I see. Maybe I was thinking `os.open()`.

I'm using a simple intro paragraph in the description now, and including the return type there instead of the shorter section above, BTW.